instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/ContextSpanProcessor.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ImplicitContextKeyed;
+import io.opentelemetry.instrumentation.api.internal.ContextSpanProcessorImpl;
+import java.util.function.BiConsumer;
+
+/**
+ * A span processing function that can be stored in context. When stored in context this function
+ * will be applied to all spans create while the context is active. Processing function is called
+ * synchronously on the execution thread, it should not throw or block the execution thread.
+ */
+public interface ContextSpanProcessor extends ImplicitContextKeyed {
```

Comment:
Can you move this class to `instrumentation-api-semconv`? (I guess it's probably the best place for incubating code?) As we want to release stable `instrumentation-api` soon, we shouldn't add new experimental utils there.

Also, there are other angles that we might want to consider in the future:
* making it possible to load SPI implementations directly from the application code; you could add a `SpanProcessor` without an extension this way;
* or going through spec with this -- there's an OTEP about `Context`-scoped attributes (https://github.com/open-telemetry/oteps/pull/207) that looks really similar to this.
If any of these get implemented at some point we'll want to deprecate and remove the `ContextSpanProcessor`.

Reply:
oh interesting I hadn't made the connection with the context-scoped attributes OTEP 👍